### DISCUSSION TITLE
Support for portable access to max file path and current working directory

### DISCUSSION THREAD

**Discussion by mcandre** [libuv/libuv#3531] [Ideas]
UNIX implementations unfortunately provide a very wide range of macros to access the max file path. libuv could help to unify all of these into a common macro. For the C programmer, it's not as simple as just `PATH_MAX` or `MAX_PATH`, but other varieties exist as well.

In terms of accessing the current working directory, getcwd works for POSIX, though Windows may do its own thing. Would be helpful to provide a common utility function there, as well.

**Comment by vtjnash** [comment]
You mean `uv_cwd`? I don't know what you are asking here

**Comment by mcandre** [comment]
Ah, uv_cwd, excellent, yes!

An ultra portable MAX path variable is still useful, however, for various other file system calls. Not even the UNIX implementations agree on what the variable should be called.

I've taken to writing my own macro for this, but would prefer that someone more detailed than me to maintain it as part of a library.

**Reply by bnoordhuis** [reply]
Something like `UV_PATH_MAX` or `int uv_max_path(void)` is problematic because, _technically_ at least, there doesn't have to be an upper bound. Callers would still need to be prepared to handle paths of arbitrary length, kind of defeating the purpose.

The second problem is that the maximum path length may depend on the file system. That is, the maximum length under `/foo` and `/bar` can be different if they're on different file systems.

**Comment by mcandre** [comment]
Granted, an fs agnostic mechanism is needed. But a basic macro would do a
lot of good.
